src/vs/workbench/contrib/welcomeGettingStarted/browser/startupPage.ts

```diff
@@ -79,14 +79,21 @@ export class StartupPageContribution implements IWorkbenchContribution {
 			if (!this.editorService.activeEditor || this.layoutService.openedDefaultEditors) {
 				const startupEditorSetting = this.configurationService.inspect<string>(configurationKey);
 
+
+				const isStartupEditorReadme = startupEditorSetting.value === 'readme';
+				const isStartupEditorUserReadme = startupEditorSetting.userValue === 'readme';
+				const isStartupEditorDefaultReadme = startupEditorSetting.defaultValue === 'readme';
+
 				// 'readme' should not be set in workspace settings to prevent tracking,
-				// but it can be set as a default (as in codespaces) or a user setting
-				const openWithReadme = startupEditorSetting.value === 'readme' &&
-					(startupEditorSetting.userValue === 'readme' || startupEditorSetting.defaultValue === 'readme');
+				// but it can be set as a default (as in codespaces or from configurationDefaults) or a user setting
+				if (isStartupEditorReadme && (!isStartupEditorUserReadme || !isStartupEditorDefaultReadme)) {
+					console.error(`Warning: 'workbench.startupEditor: readme' setting ignored due to being set somewhere other than user or default settings (user=${startupEditorSetting.userValue}, default=${startupEditorSetting.defaultValue})`);
```

Comment:
We don't want to leave console.error's in the code. If you want logs, `ILogService.warn` seems most appropriate for this